File: src/main/scala/info/crlog/higgs/ssl/SSLConfiguration.scala
package info.crlog.higgs.ssl

class SSLConfiguration {
  private var keyStorePath: String = null
  private var keyPassword: String = null
  private var keyStorePassword: String = null
  private var keyStoreType: String = "JKS"
  private var keyManagerFactoryType: String = "SunX509"
  private var trustStorePath: String = null
  private var trustStorePassword: String = null
  private var trustStoreType: String = "JKS"
  private var trustManagerFactoryType: String = "SunX509"
  private var securityProviderClass: String = "com.sun.net.ssl.internal.www.protocol"
  private var securityProtocol: String = "TLS"

  def getKeyStorePath: String = {
    return keyStorePath
  }

  def setKeyStorePath(keyStorePath: String) {
    this.keyStorePath = keyStorePath
  }

  def getKeyPassword: String = {
    return keyPassword
  }

  def setKeyPassword(keyPassword: String) {
    this.keyPassword = keyPassword
  }

  def getKeyStorePassword: String = {
    return keyStorePassword
  }

  def setKeyStorePassword(keyStorePassword: String) {
    this.keyStorePassword = keyStorePassword
  }

  def getKeyStoreType: String = {
    return keyStoreType
  }

  def setKeyStoreType(keyStoreType: String) {
    this.keyStoreType = keyStoreType
  }

  def getKeyManagerFactoryType: String = {
    return keyManagerFactoryType
  }

  def setKeyManagerFactoryType(keyManagerFactoryType: String) {
    this.keyManagerFactoryType = keyManagerFactoryType
  }

  def getTrustManagerFactoryType: String = {
    return trustManagerFactoryType
  }

  def setTrustManagerFactoryType(trustManagerFactoryType: String) {
    this.trustManagerFactoryType = trustManagerFactoryType
  }

  def getTrustStorePath: String = {
    return trustStorePath
  }

  def setTrustStorePath(trustStorePath: String) {
    this.trustStorePath = trustStorePath
  }

  def getTrustStorePassword: String = {
    return trustStorePassword
  }

  def setTrustStorePassword(trustStorePassword: String) {
    this.trustStorePassword = trustStorePassword
  }

  def getTrustStoreType: String = {
    return trustStoreType
  }

  def setTrustStoreType(trustStoreType: String) {
    this.trustStoreType = trustStoreType
  }

  def getSecurityProviderClass: String = {
    return securityProviderClass
  }

  def setSecurityProviderClass(securityProviderClass: String) {
    this.securityProviderClass = securityProviderClass
  }

  def getSecurityProtocol: String = {
    return securityProtocol
  }

  def setSecurityProtocol(securityProtocol: String) {
    this.securityProtocol = securityProtocol
  }
}

File: higgs-core/src/main/java/com/fillta/higgs/ssl/SSLConfiguration.java
package com.fillta.higgs.ssl;

public class SSLConfiguration {

	private String keyStorePath;
	private String keyPassword;
	private String keyStorePassword;
	private String keyStoreType = "JKS";
	private String keyManagerFactoryType = "SunX509";

	private String trustStorePath;
	private String trustStorePassword;
	private String trustStoreType = "JKS";
	private String trustManagerFactoryType = "SunX509";

	private String securityProviderClass = "com.sun.net.ssl.internal.www.protocol";
	private String securityProtocol = "TLS";


	public String getKeyStorePath() {
		return keyStorePath;
	}

	public void setKeyStorePath(String keyStorePath) {
		this.keyStorePath = keyStorePath;
	}

	public String getKeyPassword() {
		return keyPassword;
	}

	public void setKeyPassword(String keyPassword) {
		this.keyPassword = keyPassword;
	}

	public String getKeyStorePassword() {
		return keyStorePassword;
	}

	public void setKeyStorePassword(String keyStorePassword) {
		this.keyStorePassword = keyStorePassword;
	}

	public String getKeyStoreType() {
		return keyStoreType;
	}

	public void setKeyStoreType(String keyStoreType) {
		this.keyStoreType = keyStoreType;
	}

	public String getKeyManagerFactoryType() {
		return keyManagerFactoryType;
	}

	public void setKeyManagerFactoryType(String keyManagerFactoryType) {
		this.keyManagerFactoryType = keyManagerFactoryType;
	}

	public String getTrustManagerFactoryType() {
		return trustManagerFactoryType;
	}

	public void setTrustManagerFactoryType(String trustManagerFactoryType) {
		this.trustManagerFactoryType = trustManagerFactoryType;
	}

	public String getTrustStorePath() {
		return trustStorePath;
	}

	public void setTrustStorePath(String trustStorePath) {
		this.trustStorePath = trustStorePath;
	}

	public String getTrustStorePassword() {
		return trustStorePassword;
	}

	public void setTrustStorePassword(String trustStorePassword) {
		this.trustStorePassword = trustStorePassword;
	}

	public String getTrustStoreType() {
		return trustStoreType;
	}

	public void setTrustStoreType(String trustStoreType) {
		this.trustStoreType = trustStoreType;
	}

	public String getSecurityProviderClass() {
		return securityProviderClass;
	}

	public void setSecurityProviderClass(String securityProviderClass) {
		this.securityProviderClass = securityProviderClass;
	}

	public String getSecurityProtocol() {
		return securityProtocol;
	}

	public void setSecurityProtocol(String securityProtocol) {
		this.securityProtocol = securityProtocol;
	}
}
